INTERNSHIP/env/Lib/site-packages/pip/_internal/cli/progress_bars.py

```
@@ -0,0 +1,68 @@
+from pip._internal.utils.logging import get_indentation
+
+DownloadProgressRenderer = Callable[[Iterable[bytes]], Iterator[bytes]]
+
+
+def _rich_progress_bar(
+    iterable: Iterable[bytes],
+    *,
+    bar_type: str,
+    size: int,
+) -> Generator[bytes, None, None]:
+    assert bar_type == "on", "This should only be used in the default mode."
```

Ruff (0.12.2)

Use of `assert` detected

(S101)